File: generated_code/classes.py
from typing import List


class User:
    def __init__(self, id: int, username: str, email: str):
        
        
        self.id: int = id
        
        self.username: str = username
        
        self.email: str = email
        

    
    def create_task(self, title: str, project: Project) -> 'Task':
        pass
    

class Project:
    def __init__(self, id: int, name: str, owner: User, members: List[User]):
        
        
        self.id: int = id
        
        self.name: str = name
        
        self.owner: User = owner
        
        self.members: List[User] = members
        

    
    def add_member(self, user: User) -> None:
        pass
    
    def create_task(self, title: str, creator: User) -> 'Task':
        pass
    

class Task:
    def __init__(self, id: int, title: str, description: str, status: str, assignee: User, project: Project, comments: List[Comment]):
        
        
        self.id: int = id
        
        self.title: str = title
        
        self.description: str = description
        
        self.status: str = status
        
        self.assignee: User = assignee
        
        self.project: Project = project
        
        self.comments: List[Comment] = comments
        

    
    def change_status(self, new_status: str) -> None:
        pass
    
    def assign_to(self, user: User) -> None:
        pass
    

class Comment:
    def __init__(self, id: int, content: str, author: User, task: Task):
        
        
        self.id: int = id
        
        self.content: str = content
        
        self.author: User = author
        
        self.task: Task = task
        

    
    def edit(self, new_content: str) -> None:
        pass
    

File: generated_code/classes.cpp
#include <string>
#include <vector>

class User;
class Project;
class Task;
class Comment;


class User {
public:
    Task create_task(std::string title, Project project);
private:int id;
    std::string username;
    std::string email;
    };

class Project {
public:
    void add_member(User user);
    Task create_task(std::string title, User creator);
private:int id;
    std::string name;
    User owner;
    std::vector<User> members;
    };

class Task {
public:
    void change_status(std::string new_status);
    void assign_to(User user);
private:int id;
    std::string title;
    std::string description;
    std::string status;
    User assignee;
    Project project;
    std::vector<Comment> comments;
    };

class Comment {
public:
    void edit(std::string new_content);
private:int id;
    std::string content;
    User author;
    Task task;
    };
